airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -0,0 +1,222 @@
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_benchmark = latest_record[self.cursor_field]
+        if current_stream_state.get(self.cursor_field):
+            return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
+        return {self.cursor_field: latest_benchmark}
+
+    def _get_access_token(self) -> str:
+        """
+        Get's the access token
+        :return: access_token str
+        """
+        data = {"grant_type": "refresh_token", **self._access_token_credentials}
+        headers = {"User-Agent": "python-sp-api-0.6.2", "content-type": "application/x-www-form-urlencoded;charset=UTF-8"}
+        res = requests.post("https://api.amazon.com/auth/o2/token", data, headers)
+        return res.json()["access_token"]
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return {
+            "host": "sellingpartnerapi-na.amazon.com",
+            "user-agent": "python-sp-api-0.6.2",
+            "x-amz-access-token": self._get_access_token(),
+            "x-amz-date": pendulum.now("utc").strftime("%Y%m%dT%H%M%SZ"),
+            "content-type": "application/json",
+        }
+
+    def _create_prepared_request(
+        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None, auth: AWSSigV4 = None
+    ) -> requests.PreparedRequest:
+        args = {"method": self.http_method, "url": self.url_base + path, "headers": headers, "params": params, "auth": auth}
+
+        if self.http_method.upper() == "POST":
+            args["json"] = json
+
+        return self._session.prepare_request(requests.Request(**args))
+
+    def read_records(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {}
+        pagination_complete = False
+
+        next_page_token = None
+        while not pagination_complete:
+            request_headers = self.request_headers(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
+            request = self._create_prepared_request(
+                path=self.path(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
+                headers=dict(request_headers),
+                params=self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
+                json=self.request_body_json(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
+                auth=self.authenticator,
```

Comment:
Since you copy-pasted the whole body of `read_records` too, maybe we should add this directly to `HttpStream`? @keu wdyt?
Add method `request_auth()` which will return value for `auth` in `_create_prepared_request()`. So the code will look something like this:

```
class HttpStream(Stream, ABC):
    
    ...

    def request_auth(self, **kwargs):
        pass

    def _create_prepared_request(
        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None, auth: Any = None
    ) -> requests.PreparedRequest:
        args = {"method": self.http_method, "url": self.url_base + path, "headers": headers, "params": params, "auth": auth}

        if self.http_method.upper() == "POST":
            # TODO support non-json bodies
            args["json"] = json

        return self._session.prepare_request(requests.Request(**args))


    def read_records(
        self,
        sync_mode: SyncMode,
        cursor_field: List[str] = None,
        stream_slice: Mapping[str, Any] = None,
        stream_state: Mapping[str, Any] = None,
    ) -> Iterable[Mapping[str, Any]]:
            
            ...

            request = self._create_prepared_request(
                path=self.path(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
                headers=dict(request_headers),
                params=self.request_params(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
                json=self.request_body_json(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token),
                auth=self.request_auth(...),
            )
            request_kwargs = self.request_kwargs(stream_state=stream_state, stream_slice=stream_slice, next_page_token=next_page_token)
            
            ...



class AspStream(HttpStream, ABC):
    
    ...

    def request_auth(self, **kwargs):
        return self.authenticator

```